LongLookupTables/get_long_lookup_tables.py

```diff
@@ -0,0 +1,136 @@
+
+        is_small (bool, optional): whether to run a smaller verson of the task.
+            Used for getting less statistically significant results.
+        is_mini (bool, optional): whether to run a smaller verson of the task.
+            Used for testing purposes.
+        longer_repeat (int, optional): number of longer test sets. 
+            - note if data is already generated with a certain longer repeat 
+            - If a longer repeat is called then the return paths will not exist
+            - In this case either delete the data folder from the specific LongLookup set
+            - Or implement a check to extend it and regenerate with higher longer_repeat
+
+    Returns:
+        task arguments (to be passed to instantiate a Task Object)
+    """
+
+    print("Getting Lookup Table")
+    name = name.lower()
+    lookup_tables_dir_path = os.path.join(dir_path, name2dir[name])
+    if not os.path.isdir(lookup_tables_dir_path):
+        raise NotImplementedError(
+            "Folder at {} does not exist".format(lookup_tables_dir_path))
+
+    generation_arguments_path = os.path.join(
+        lookup_tables_dir_path, 'generation_arguments.txt')
+    if not os.path.isfile(generation_arguments_path):
+        raise NotImplementedError(
+            "Generation Arguments .txt Missing in Table Lookup Folder \
+             - Cannot Generate Table")
+
+    lookup_tables_data_dir_path = os.path.join(lookup_tables_dir_path, "data")
+
+    if not os.path.isdir(lookup_tables_data_dir_path):
+        print("Data not present for {}".format(name))
+        print("Generating Dataset")
+        make_long_lookup_tables(
+            lookup_tables_data_dir_path, generation_arguments_path)
+
+    # Get default params from json
+    # - these are not required but offer recommendation on default params
+    default_params = get_default_params(lookup_tables_dir_path)
+
+    # Update the defauls params if task is small /mini
+    if default_params is not None:
+        if is_small:
+            default_params["task_defaults"]["k"] = 1
+        if is_mini:
+            default_params["task_defaults"]["k"] = 1
+            default_params["task_defaults"]["batch_size"] = 128
+            default_params["task_defaults"]["patience"] = 2
+            default_params["task_defaults"]["epochs"] = 3
+            default_params["task_defaults"]["n_attn_plots"] = 1
+
+    train_file = "train"
+    valid_file = "validation"
+    test_files = flatten(["heldout_inputs", "heldout_compositions",
+                          "heldout_tables",
+                          "new_compositions", repeat(
+                              "longer_seen", longer_repeat),
+                          repeat("longer_incremental", longer_repeat),
+                          repeat("longer_new", longer_repeat)])
+
+    return (name, lookup_tables_data_dir_path,
+            train_file, valid_file, test_files, default_params)
+
+
+# Helper Functions
```

Comment:
For me all these functions should be in a helper.py

Reply:
I've moved them to the utils folder in a `helper.py` file since they might be useful to other tasks as well.